app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,34 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments)
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+
+    if @post.save
+      @post.update_user_post_counter(params[:user_id])
+      redirect_to user_post_path(@user.id)
+      flash[:notice] = 'Your comment was successfully created'
+    else
+      redirect_to new_user_post_path(@user.id)
+      flash[:notice] = 'An error has occurred while creating the post'
+    end
+  end
```

Comment:
- [ ] While creating a new post it raises an error in the image below, So please fix that.

![image](https://user-images.githubusercontent.com/68796232/193051200-b26523c0-ea27-4d4a-a1a0-6cc45b3b18ff.png)